### ISSUE TITLE
Special characters such as Scandinavian characters are not supported on file names

### ISSUE THREAD

**Issue by TimoRiikonen** [Azure-Samples/azure-search-openai-demo#334]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ X ] bug report -> please search issues before submitting
- [ X ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> Upload file that has chars äåö in the file name.

### Any log messages given by the failure
> Invalid document key: 'ISO13485-Koulutus_sisäiset_auditoijat1_pdf-49'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check.\t(InvalidDocumentKey) 50: Invalid document key: 'ISO13485-Koulutus_sisäiset_auditoijat1_pdf-50'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check.\n\tCode: 

### Expected/desired behavior
> Scandinavian characters have been valid in directory and file names for several decades. 
System should convert file names URL-safe instead of requesting users to do that.

### OS and Version?
> Windows 11

### azd version?
> 1.0.2

### Versions
>June 7, 2023

### Mention any other details that might be useful
I haven't tested allowUnsafeKeys setting. Would file links work after that still?

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by tonybaloney** [comment]
The keys should be base64 encoded, I'll have a look at this. 
This error would apply to a large number of other filenames with UTF-8/16 characters

**Comment by tonybaloney** [comment]
Did this error occur when running `prep docs.py` / or deploying? Please can you paste the full text of the output

**Comment by TimoRiikonen** [comment]
I got the error when uploading files.
Other files were indexed except those with special characters on the file name.

Current Credentials <azure.identity._credentials.default.DefaultAzureCredential object at 0x0000013DBCF41D10>
Uploading file Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf to 1687240437
Uploaded file to /tmp\1687240437\Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf
Indexing /tmp\1687240437\Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf
        Uploading blob for page 0 -> Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf-0.pdf
        Uploading blob for page 1 -> Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf-1.pdf
        Uploading blob for page 2 -> Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf-2.pdf
        Uploading blob for page 3 -> Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf-3.pdf
        Uploading blob for page 4 -> Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf-4.pdf
        Uploading blob for page 5 -> Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf-5.pdf
        Uploading blob for page 6 -> Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf-6.pdf
        Uploading blob for page 7 -> Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf-7.pdf
Uploading blobs for /tmp\1687240437\Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf -> None
Creating sections for /tmp\1687240437\Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf -> <generator object create_sections at 0x0000013DEB94FBE0>
Ensuring search index gptkbindex exists
Search index gptkbindex already exists
Indexing sections from 'Asiakastyytyväisyyskysely - Kontaktien lisääminen CRMn projektille 1.pdf' into search index 'gptkbindex'
Splitting '<pypdf._page._VirtualList object at 0x0000013DEB87A710>' into sections
[{'success': False, 'message': "(InvalidName) The request is invalid. Details: actions : 0: Invalid document key: 'Asiakastyytyväisyyskysely_-_Kontaktien_lisääminen_CRMn_projektille_1_pdf-0'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check. 1: Invalid document key: 'Asiakastyytyväisyyskysely_-_Kontaktien_lisääminen_CRMn_projektille_1_pdf-1'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check.\nCode: InvalidName\nMessage: The request is invalid. Details: actions : 0: Invalid document key: 'Asiakastyytyväisyyskysely_-_Kontaktien_lisääminen_CRMn_projektille_1_pdf-0'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check. 1: Invalid document key: 'Asiakastyytyväisyyskysely_-_Kontaktien_lisääminen_CRMn_projektille_1_pdf-1'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check.\nException Details:\t(InvalidDocumentKey) 0: Invalid document key: 'Asiakastyytyväisyyskysely_-_Kontaktien_lisääminen_CRMn_projektille_1_pdf-0'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check. Parameters: actions\n\tCode: InvalidDocumentKey\n\tMessage: 0: Invalid document key: 'Asiakastyytyväisyyskysely_-_Kontaktien_lisääminen_CRMn_projektille_1_pdf-0'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check. Parameters: actions\t(InvalidDocumentKey) 1: Invalid document key: 'Asiakastyytyväisyyskysely_-_Kontaktien_lisääminen_CRMn_projektille_1_pdf-1'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check.\n\tCode: InvalidDocumentKey\n\tMessage: 1: Invalid document key: 'Asiakastyytyväisyyskysely_-_Kontaktien_lisääminen_CRMn_projektille_1_pdf-1'. Keys can only contain letters, digits, underscore (_), dash (-), or equal sign (=). If the keys in your source data contain other characters, we recommend encoding them with a URL-safe version of Base64 before uploading them to your index. If that is not an option, you can add the 'allowUnsafeKeys' query string parameter to disable this check."}]

**Comment by tonybaloney** [comment]
This issue was fixed in #418
